File: problems/119.pascal's-triangle-II/main.py
from typing import List

# Gì đó ở đây, chưa biết được

# Viết code ở đây
class Solution(object):
    def generate(self, numRows):
        res = [[1]]

        for _ in range(numRows - 1):
            dummy_row = [0] + res[-1] + [0]
            row = []
            print(dummy_row)

            for i in range(len(res[-1]) + 1):
                row.append(dummy_row[i] + dummy_row[i+1])
            res.append(row)
        
        return res

# ==========================
# Test local
# ==========================
if __name__ == "__main__":
    sol = Solution()
    # Viết input ở đây
    numRows = 5
    res = sol.generate(numRows)
    print(res)

# Mỗi lần thêm 2 số 0 vào đầu và cuối thì khi cộng 2 số liền kề, ta được kết quả đúng cho ô dưới

File: problems/119.pascal's-triangle-II/main.cpp
#include <bits/stdc++.h>
#define Shelter main
using namespace std;

// -------------------------Struct here------------------------

// ------------------------------------------------------------

// ------------------------Solution here-----------------------
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<int> res(1, 1);

        for (int i = 1; i <= rowIndex; i++) {
            vector<int> next;
            next.push_back(0); for (int j = 0; j < res.size(); j++) next.push_back(res[j]); next.push_back(0);
            res.clear();

            for (int j = 0; j < i; j++) {
                res.push_back(next[j] + next[j+1]);
            }
            res.push_back(1);
        }

        return res;
    }
};
// ------------------------------------------------------------

#ifndef ONLINE_JUDGE
Shelter() {
    Solution sol;
    // --------------------Input, Output here----------------------
    int numRows = 5;
    vector<int> res = sol.getRow(numRows);
    for (int i = 0; i < numRows; i++) {
        cout << res[i] << " ";
    }
    // ------------------------------------------------------------
}
#endif

/* Nháp ở dưới đây
    - Làm giống bài 118 thì nó bị phí space, vì vậy chúng ta sẽ làm theo cách của bài 118 nhưng là bản python

    -> Làm xong thấy không tiết kiệm cho lắm nhỉ
*/